src/pages/DatasetRegistration.js

```diff
@@ -163,15 +216,37 @@ class DatasetRegistration extends Component {
   };
 
   validateDatasetName(name) {
+    let oldDataset = this.state.updateDataset;
     let datasets = this.state.allDatasetNames;
-    let val = !fp.contains(name, datasets);
-    return val;
+    // create dataset case
+    if (fp.isEmpty(oldDataset)) {
+      return !fp.contains(name, datasets);
+    }
+    // update dataset case (include old dataset name as valid)
+    else {
+      let updateDatasetName = fp.find(p => p.propertyName === "Dataset Name", this.state.updateDataset.properties).propertyValue;
+
+      return (name === updateDatasetName || !fp.contains(name, datasets));
+    }
   };
 
   showDatasetNameErrors(name, showValidationMessages) {
     if (fp.isEmpty(name)) {
       return showValidationMessages ? 'form-control required-field-error' : 'form-control';
     }
+    // if there is a name loaded in because this is an update
+    if (!fp.isEmpty(this.state.updateDataset)) {
+      let updateDatasetName = fp.find(p => p.propertyName === "Dataset Name", this.state.updateDataset.properties).propertyValue;
+      let equalsOriginal = (name === updateDatasetName)
```

Comment:
[codacy nit:](https://app.codacy.com/gh/DataBiosphere/duos-ui/pullRequest?prid=6449150)
```suggestion
      let equalsOriginal = (name === updateDatasetName);
```